edge-hub/watchdog/linux/src/main.rs

```diff
@@ -0,0 +1,134 @@
+extern crate nix;
+
+use std::{thread};
+use std::io::Error;
+use std::process::{Command, Stdio, Child};
+use std::sync::{Arc};
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::time::Duration;
+use nix::unistd::Pid;
+use nix::sys::signal::{self, Signal};
+use signal_hook::{iterator::Signals, SIGTERM};
+use tracing::{info, error};
+use tracing_subscriber;
+
+fn main() -> Result<(), Error> {
+    init_logging();
+
+    let has_received_sigterm = match register_sigterm_listener() {
+        Ok(has_received_sigterm) => has_received_sigterm,
+        Err(e) => {
+            error!("Failed to register sigterm listener. Shutting down.");
+            return Err(e)
+        }
+    };
+
+    let mut edgehub = Command::new("dotnet")
+            .arg("/app/Microsoft.Azure.Devices.Edge.Hub.Service.dll")
+            .stdout(Stdio::inherit())
+            .spawn()
+            .expect("Failed to execute Edge Hub process");
+    
+    let mut broker = Command::new("/usr/local/bin/mqttd")
+            .stdout(Stdio::inherit())
+            .spawn()
+            .expect("failed to execute MQTT broker process");
+
+    info!("Launched Edge Hub process with pid {:?}", edgehub.id());
+    info!("Launched MQTT Broker process with pid {:?}", broker.id());
+
+    wait_for_sigterm_or_processes_to_shutdown(&mut edgehub, &mut broker, has_received_sigterm);
+
+    shutdown(&mut edgehub, &mut broker);
+
+    info!("Exiting");
+    Ok(())
+}
+
+fn init_logging() {
+    let subscriber = tracing_subscriber::fmt::Subscriber::builder()
+        .with_max_level(tracing::Level::INFO)
+        .finish();
+    let _ = tracing::subscriber::set_global_default(subscriber);
+
+    info!("Starting watchdog");
+}
+
+fn register_sigterm_listener() -> Result<Arc<std::sync::atomic::AtomicBool>, Error> {
+    let signals = Signals::new(&[SIGTERM])?;
+    let has_received_sigterm = Arc::new(AtomicBool::new(true));
+    let sigterm_listener = has_received_sigterm.clone();
+    thread::spawn(move || {
+        for _sig in signals.forever() {
+            info!("Received SIGTERM for watchdog");
+            sigterm_listener.store(false, Ordering::Relaxed);
+        }
+    });
+
+    Ok(has_received_sigterm)
+}
+
+fn wait_for_sigterm_or_processes_to_shutdown(mut edgehub: &mut Child, mut broker: &mut Child, has_received_sigterm: Arc<std::sync::atomic::AtomicBool>) {
+    let mut is_edgehub_running = is_child_process_running(&mut edgehub);
+    let mut is_broker_running = is_child_process_running(&mut broker);
+    while has_received_sigterm.load(Ordering::Relaxed) && is_edgehub_running && is_broker_running {
```

Comment:
Ordering::Relaxed is the appropriate ordering since we only care about atomic operations.
https://doc.rust-lang.org/std/sync/atomic/enum.Ordering.html#variant.Relaxed